src/mon/Monitor.cc

```diff
@@ -2958,7 +2959,13 @@ void Monitor::get_cluster_status(stringstream &ss, Formatter *f)
     mgrstatmon()->print_summary(f, NULL);
     f->close_section();
     f->open_object_section("fsmap");
-    mdsmon()->get_fsmap().print_summary(f, NULL);
+    const FSMap *fsmapp = &mdsmon()->get_fsmap();
+    if (!fsids.empty()) {
+      FSMap map = *fsmapp;
+      map.filter(fsids);
+      fsmapp = &map;
```

Comment:
This code is not safe. `map` goes out of scope and is destructed after this statement.

Reply:
Fixed it. Let me know if it looks good - https://github.com/ceph/ceph/pull/32581/files#diff-d4e1ba36dd08617ea271ab2be3dbcb5bR2963